oxidation/libparsec/crates/crypto/src/rustcrypto/sign.rs

```diff
@@ -129,22 +129,20 @@ impl VerifyKey {
             return Err(CryptoError::Signature);
         }
         self.verify_with_signature(
-            &signed[..Signature::BYTE_SIZE],
+            signed[..Signature::BYTE_SIZE]
+                .try_into()
+                .expect("Unreacheable, because it's the correct size"),
```

Comment:
This is not the same behavior than for sodiumoxide (which retuns a `Err(CryptoError::SignatureVerification)`)

And there should be an additional test which tries to do the verify on an invalid signature (cf. https://github.com/RustCrypto/signatures/blob/fd78bf79cb7188d908846c15965a57adab638dec/ed25519/src/lib.rs#L314 for building an invalid signature)

Reply:
Can we do it in another PR ? Because I added this method only to build with `sodiumoxide`, and this PR is already big

Reply:
I don't think we should play fool with this PR given how critical the modified parts are.